src/main/perl/RuleBasedEditor.pm

```diff
@@ -0,0 +1,1023 @@
+
+
+#########################################################
+# Constants use to format lines in configuration files. #
+# These constants are exported.                         #
+#########################################################
+
+#  LINE_FORMAT_xxx: general syntax of the line (key/val format)
+# LINE_FORMAT_SH_VAR:         key=val (e.g. SH shell family)
+# LINE_FORMAT_ENV_VAR:        export key=val (e.g. SH shell family)
+# LINE_FORMAT_KEY_VAL:        key val (e.g. Xrootd, Apache)
+# LINE_FORMAT_KEY_VAL_SETENV: setenv key val  (used by Xrootd in particular)
+# LINE_FORMAT_KEY_VAL_SET:    set key val  (used by Xrootd in particular)
+use enum qw(LINE_FORMAT_SH_VAR=1
+            LINE_FORMAT_ENV_VAR
+            LINE_FORMAT_KEY_VAL
+            LINE_FORMAT_KEY_VAL_SETENV
+            LINE_FORMAT_KEY_VAL_SET
+           );
+
+# LINE_VALUE_xxx: how to interpret the configuration value
+# LINE_VALUE_INSTANCE_PARAMS is specific to Xrootd.
+use enum qw(LINE_VALUE_AS_IS
+            LINE_VALUE_BOOLEAN
+            LINE_VALUE_ARRAY
+            LINE_VALUE_HASH_KEYS
+            LINE_VALUE_STRING_HASH
+            LINE_VALUE_INSTANCE_PARAMS
+           );
+
+# LINE_VALUE_OPT_xxx: options for rendering the value
+# (mainly apply to lists and dictionnaries)
+use enum qw(BITMASK: LINE_VALUE_OPT_SINGLE
+                     LINE_VALUE_OPT_UNIQUE
+                     LINE_VALUE_OPT_SORTED
+           );
+
+# Internal constants
+Readonly my $LINE_FORMAT_DEFAULT => LINE_FORMAT_SH_VAR;
+Readonly my $LINE_QUATTOR_COMMENT => "\t\t# Line generated by Quattor";
+Readonly my $LINE_OPT_DEF_REMOVE_IF_UNDEF => 0;
+Readonly my $LINE_OPT_DEF_ALWAYS_RULES_ONLY => 0;
+Readonly my $RULE_CONDITION_ALWAYS => 'ALWAYS';
+Readonly my $RULE_OPTION_SET_GLOBAL => 'GLOBAL';
+
+
+# Export constants used to build rules
+Readonly my @RULE_CONSTANTS => qw(LINE_FORMAT_SH_VAR
+                                  LINE_FORMAT_ENV_VAR
+                                  LINE_FORMAT_KEY_VAL
+                                  LINE_FORMAT_KEY_VAL_SETENV
+                                  LINE_FORMAT_KEY_VAL_SET
+                                  LINE_VALUE_AS_IS
+                                  LINE_VALUE_BOOLEAN
+                                  LINE_VALUE_INSTANCE_PARAMS
+                                  LINE_VALUE_ARRAY
+                                  LINE_VALUE_HASH_KEYS
+                                  LINE_VALUE_STRING_HASH
+                                  LINE_VALUE_OPT_SINGLE
+                                  LINE_VALUE_OPT_UNIQUE
+                                  LINE_VALUE_OPT_SORTED
+                                 );
+our @EXPORT_OK;
+our %EXPORT_TAGS;
+push @EXPORT_OK, @RULE_CONSTANTS;
+$EXPORT_TAGS{rule_constants} = \@RULE_CONSTANTS;
+
+
+# Backup file extension
+Readonly my $BACKUP_FILE_EXT => ".old";
+
+
+=pod
+
+=head1 DESCRIPTION
+
+This module implements a rule-based editor. It is a subclass of the B<CAF::FileEditor>
```

Comment:
`It extends `B<CAF::FileEditor>.` (Not sure what `B` does, but `L` is supposed to link